### PR TITLE
refactor: add SKU field in the form to create product

### PR DESCRIPTION
**Pull request by Blossom1-chike**
## Description
[Please put a description of the changes this branch introduces here.]
add form field for SKU when creating a product
add SKU to the payload sent to the api when creating a product

## To Test
[Please write instructions for testing your changes locally here. Include any data setup, tests, urls, and potential snags in local testing.]

## Checklist
- [ ] I have looked over the diffs.
- [ ] There are no errors/failing tests from my changes on this branch.
- [ ] I have requested a review on this PR and re-assigned the Jira ticket.

## Screenshots
[Any screenshots of the changes can go here.]

## Extra Notes
[Please include any extra notes you may have about the changes here.]

### CODE REVIEW

File: src/components/products-changes/index.js [hunk 1]
```diff
@@ -124,6 +124,28 @@ const ProductChanges = ({
...
                 </div>
+                <div className=" md:w-[327px] w-[50%]">
+                  <div className="mb-8 text-start w-[327px] z-0">
+                    <InputComponent
+                      inputType="text"
+                      label="SKU"
+                      fieldName="sku"
+                      placeholder="Enter"
+                      value={productInfo?.sku}
+                      patternValue={"^AFR-[a-zA-Z0-9]+"}
```
Comment (ObiTam) [inline src/components/products-changes/index.js:135]: Hi blossom, 

It doesn't have to start with AFR. Lets just add a minimum limit of 5 characters - alphanumeric.

**Review by ObiTam — CHANGES_REQUESTED** [review]
Hi Blossom, 

I think the choice for custom SKUs should be at store level. If a Store chooses to use custom SKUs then all products must have custom SKUs alternatively the SKUs will be user generated.  

That means we would add a toggle on the store profile info that is not editable. "use Custom SKU". 

@Eshy10 what do you think?

Please refactor.

**Comment by Eshy10** [comment]
> Hi Blossom,
> 
> I think the choice for custom SKUs should be at store level. If a Store chooses to use custom SKUs then all products must have custom SKUs alternatively the SKUs will be user generated.
> 
> That means we would add a toggle on the store profile info that is not editable. "use Custom SKU".
> 
> @Eshy10 what do you think?
> 
> Please refactor.

@ObiTam this works

**Comment by Blossom1-chike** [comment]
> Hi Blossom,
> 
> I think the choice for custom SKUs should be at store level. If a Store chooses to use custom SKUs then all products must have custom SKUs alternatively the SKUs will be user generated.
> 
> That means we would add a toggle on the store profile info that is not editable. "use Custom SKU".
> 
> @Eshy10 what do you think?
> 
> Please refactor.

this sounds like the option to use custom SKUs is accross the store and not the product, so for a store that already has products with generated SKUs and i want to use custom SKUs, what happens?

**Comment by ObiTam** [comment]
> > Hi Blossom,
> > I think the choice for custom SKUs should be at store level. If a Store chooses to use custom SKUs then all products must have custom SKUs alternatively the SKUs will be user generated.
> > That means we would add a toggle on the store profile info that is not editable. "use Custom SKU".
> > @Eshy10 what do you think?
> > Please refactor.
> 
> this sounds like the option to use custom SKUs is accross the store and not the product, so for a store that already has products with generated SKUs and i want to use custom SKUs, what happens? @ObiTam

lack of custom skus falls back to generated sku.

**Review by ObiTam — APPROVED** [review]

**Comment by ObiTam** [comment]
@Eshy10 @Davidosky007 
This is pending till review of backend PR.

**Pull request merged**